addons/website_sale_wishlist/static/src/js/website_sale.js

```diff
@@ -13,4 +22,276 @@ WebsiteSale.include({
+    },
+
+    /**
+     * @private
+     */
+    _updateWishlistView: function () {
+        const $wishButton = $('.o_wsale_my_wish');
+        if ($wishButton.hasClass('o_wsale_my_wish_hide_empty')) {
+            $wishButton.toggleClass('d-none', !this.wishlistProductIDs.length);
+        }
+        $wishButton.find('.my_wish_quantity').text(this.wishlistProductIDs.length);
+    },
+
+    /**
+     * @private
+     */
+    _removeWish: function (e, deferred_redirect) {
+        let tr = $(e.currentTarget).parents('tr');
+        let wish = tr.data('wish-id');
+        let product = tr.data('product-id');
+        let self = this;
+
+        this._rpc({
+            route: '/shop/wishlist/remove/' + wish,
+        }).then(function () {
+            $(tr).hide();
+        });
+
+        this.wishlistProductIDs = _.without(this.wishlistProductIDs, product);
+        if (this.wishlistProductIDs.length === 0) {
+            if (deferred_redirect) {
+                deferred_redirect.then(function () {
+                    self._redirectNoWish();
+                });
+            }
+        }
+        this._updateWishlistView();
+    },
+
+    /**
+     * @private
+     */
+    _addOrMoveWish: function (e) {
+        let $navButton = $('header .o_wsale_my_cart').first();
+        let tr = $(e.currentTarget).parents('tr');
+        let product = tr.data('product-id');
+        $('.o_wsale_my_cart').removeClass('d-none');
+        wSaleUtils.animateClone($navButton, tr, 25, 40);
+
+        if ($('#b2b_wish').is(':checked')) {
+            return this._addToCart(product, tr.find('add_qty').val() || 1);
+        } else {
+            let adding_deffered = this._addToCart(product, tr.find('add_qty').val() || 1);
+            this._removeWish(e, adding_deffered);
+            return adding_deffered;
+        }
+    },
+
+    /**
+     * @private
+     */
+    _addToCart: function (productID, qty) {
+        const $tr = this.$(`tr[data-product-id="${productID}"]`);
+        const productTrackingInfo = $tr.data('product-tracking-info');
+        if (productTrackingInfo) {
+            productTrackingInfo.quantity = qty;
+            $tr.trigger('add_to_cart_event', [productTrackingInfo]);
+        }
+        return this._rpc({
+            route: "/shop/cart/update_json", params: this._getCartUpdateJsonParams(productID, qty),
+        }).then(function (resp) {
+            if (resp.warning) {
+                if (!$('#data_warning').length) {
+                    $('.wishlist-section').prepend('<div class="mt16 alert alert-danger alert-dismissable" role="alert" id="data_warning"></div>');
+                }
+                let cart_alert = $('.wishlist-section').parent().find('#data_warning');
+                cart_alert.html('<button type="button" class="close" data-dismiss="alert" aria-hidden="true">&times;</button> ' + resp.warning);
+            }
+            $('.my_cart_quantity').html(resp.cart_quantity || '<i class="fa fa-warning" /> ');
```

Comment:
For security:
The code was moved, it already existed in the following file: `addons/website_sale_wishlist/static/src/js/website_sale_wishlist.js`